src/discord/index.js

```diff
@@ -107,22 +107,25 @@ export default class Bot extends EventEmitter {
         case '!add':
         case '!play':
         case '!queue': {
-          const res = radio.addSong(args[1], message.author);
-          if (!res) {
-            break;
-          }
-          res.on('error', err => {
-            message.reply(`I couldn't add that! The error was: \`${err.message}\``);
-            console.error(err.stack);
-          });
-          res.on('done', song => {
-            const duration = timeStr(song.duration);
-
-            message.reply(
-              `queueing ${song.title} [${duration}] uploaded by ${song.uploader.name}` + '' // ` (${rating_str}★ / ${views_str} views)`
-            );
-
-            setTopic();
+          radio.addSong(args[1], message.author).then(result => {
+            if (result) {
```

Comment:
can we ever not-get a result here?

Reply:
if so, remember that we will prefer a
```js
if (/* error condition */) {
  // handle it
  return; // bail
}
```
rather than wrapping the entire body of a thing
- readability
- simplifies flow
- less indentation / brace waterfall